event_reporter/reporter/event_payload.go

```diff
@@ -154,6 +155,43 @@ func getResourceEventPayload(
 	return &events.Event{Payload: payloadBytes}, nil
 }
 
+func getResourceSourceRepoUrl(
+	rr *ReportedResource,
+	reportedEntityParentApp *ReportedEntityParentApp,
+) string {
+	specCopy := reportedEntityParentApp.app.Spec.DeepCopy()
+
+	specCopy.Sources = reportedEntityParentApp.app.Status.Sync.ComparedTo.Sources
+	specCopy.Source = reportedEntityParentApp.app.Status.Sync.ComparedTo.Source.DeepCopy()
+
+	if specCopy.HasMultipleSources() {
+		if rr.appSourceIdx == -1 {
```

Comment:
lets move this condition to ReportedResource as dedicated func, you use it in multiple places